makefiles/features.modules.mk

```
@@ -0,0 +1,35 @@
+# Add modules implementing used features
+#
+# This is done after the regular dependency resolution in Makefile.dep, as
+# feature resolution depends on the used modules. As these modules however have
+# no dependencies(except for periph_common), no dependency resolution is needed.
+
+PERIPH_FEATURES := $(filter periph_%,$(FEATURES_USED))
+# all periph features correspond to a periph submodule
+# FEATURES_USED is defined in Makefile.features
+USEMODULE += $(PERIPH_FEATURES)
+
+# select cpu_check_address pseudomodule if the corresponding feature is used
+USEMODULE += $(filter cpu_check_address, $(FEATURES_USED))
+
+# include periph_common if any periph_* driver is used
+ifneq (,$(filter periph_%, $(USEMODULE)))
+  USEMODULE += periph_common
+endif
+
+# set all USED periph_% init modules as DEFAULT_MODULE
+ifneq (,$(filter periph_init, $(USEMODULE)))
+    # To enable adding periph_% modules through the environment we cant used
+    # PERIPH_FEATURES since the MODULE might be have added directly as USEMODULE
+    PERIPH_MODULES_NO_INIT = periph_init periph_common
```

```suggestion
    PERIPH_MODULES_NO_INIT = periph_init% periph_common
```

With the wildcard it would be more bullet  proof, e.g. when some does `USEMODULE="periph_init_i2c periph_i2c make`.

---

This was actually necessary, I first ran `save_all_dependencies_resolution_variables.sh` on the first version of the PR and although there was no `periph_init_init` or `periph_init_common`, there where many `periph_ini_init_gpio`.

This was because I didn't use `:=` for `PERIPH_MODULES` so they were resolved at the end and include `init` modules. I fixed this by including your suggestion as well as using immediate expansion.

---

I'll re run the script